File: 02-04-2025_Maximum_Value_of_an_Ordered_Triplet_1.py
def maximum_triplet_value(nums):
    max_val = float('-inf')
    nums_size = len(nums)

    for i in range(nums_size - 2):
        for j in range(i + 1, nums_size - 1):
            for k in range(j + 1, nums_size):
                tmp = (nums[i] - nums[j]) * nums[k]
                if tmp > max_val:
                    max_val = tmp

    return max_val if max_val >= 0 else 0

File: 02-04-2025_Maximum_Value_of_an_Ordered_Triplet_1.cpp
long long maximumTripletValue(vector<int>& nums) {
    long long maxVal = LLONG_MIN;
    int numsSize = nums.size();
    
    for(int i = 0; i < numsSize - 2; ++i) {
        for(int j = i + 1; j < numsSize - 1; ++j) {
            for(int k = j + 1; k < numsSize; ++k) {
                long long tmp = (long long)(nums[i] - nums[j]) * nums[k];
                if(tmp > maxVal) {
                    maxVal = tmp;
                }
            }
        }
    }
    
    return maxVal < 0 ? 0 : maxVal;
}
